File: java/src/main/java/facade/account/CheckingAccount.java
package facade.account;

public class CheckingAccount extends Account {
    private static final long DEPOSIT_LIMIT = 5_000_000;

    public CheckingAccount(long balance) {
        super(balance);
    }

    @Override
    public void deposit(long amount) {
        if (amount > DEPOSIT_LIMIT) {
            throw new IllegalArgumentException("1회 입금 한도 초과");
        }

        addBalance(amount);
    }

    @Override
    public void withdraw(long amount) {
        minusBalance(amount);
    }
}

File: kotlin/src/main/kotlin/facade/account/CheckingAccount.kt
package facade.account

class CheckingAccount(balance: Long) : Account(balance) {
    companion object {
        const val DEPOSIT_LIMIT: Long = 5_000_000
    }

    override fun deposit(amount: Long) {
        if (amount > DEPOSIT_LIMIT) {
            throw IllegalArgumentException("1회 입금 한도 초과")
        }

        addBalance(amount)
    }

    override fun withdraw(amount: Long) {
        minusBalance(amount)
    }
}